WasteWar/Assets/Scripts/Grid/GameLogicGrid.cs

```diff
@@ -0,0 +1,38 @@
+﻿using UnityEngine;
+
+
+public class GameLogicGrid
+{
+    private readonly Terrain terrain;
+    private Mesh cells = new Mesh();
+    public float CellSize { get; } = 1f;
+    public int[,] Elements { get; set; }
+    Vector3 terrainPos;
+    Vector3 terrainSize;
+
+    public GameLogicGrid(Terrain terrain)
+    {
+        this.terrain = terrain;
+        terrainPos = terrain.transform.position;
+        terrainSize = terrain.GetComponent<Terrain>().terrainData.size;
```

Comment:
you don't need getcomponent i guess

Reply:
this is work in progress so it shouldn't be considered